src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        objects = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= capacity) {
+            capacity = grow();
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size() || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist: " + index);
+        }
+        if (size == capacity) {
+            capacity = grow();
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        System.arraycopy(objects, index, objects, index + 1, size - index);
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return (T) objects[index];
+    }
+
+    private void indexCheck(int index) {
```

Comment:
let's move private method after all public
```suggestion
    private void checkIndex(int index) {
```